### PR TITLE
refined container logic of TypeParameterDeclarations

### PR DESCRIPTION
**Pull request by mlangkabel**
* implemented getContainer method for all TypeParameterDeclarations
* allowed all TypeParameterDeclaration to have a ConstructorDeclaration as Container

### CODE REVIEW

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionTypeParameter.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {
         int result = typeVariable.hashCode();
-        result = 31 * result + (declaredOnClass ? 1 : 0);
+        result = 31 * result + (declaredOnType() ? 1 : 0);
```
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionTypeParameter.java]: maybe we can use the container here

Comment (mlangkabel) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionTypeParameter.java]: That sounds reasonable. However, I'm quite unfamiliar with Java HashCodes so I have to ask: Are you thinking of just adding the hashcode of the container to the `result * 31`?

Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionTypeParameter.java]: Yes, something like:
```
result = 31 * result + container.hashCode();
```

Comment (mlangkabel) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionTypeParameter.java]: Ok, I'll change that and merge afterwards. Thanks for reviewing :)

**Comment by ftomassetti** [comment]
Great, thanks. I plan to do a release soon after the recent PRs

**Comment by mlangkabel** [comment]
That sounds good! I have another local branch I'd like to push today and I'm planning on doing some more fixes this week. Just want you to know!

**Comment by ftomassetti** [comment]
Good, then I can wait and do the release on thursday. What is not included can always go in another release within a couple of weeks.

**Pull request merged**